pkg/kev/skaffold.go

```diff
@@ -262,6 +284,88 @@ func (s *SkaffoldManifest) AdditionalProfiles() {
 	}
 }
 
+// SetBuildArtifacts detects build artifacts from the current project and adds `build` section to the manifest
+func (s *SkaffoldManifest) SetBuildArtifacts(analysis *Analysis) error {
+
+	// don't set build artefacts if no analysis available
+	if analysis == nil {
+		return nil
+	}
+
+	buildArtifacts := map[string]string{}
+
+	for _, d := range analysis.Dockerfiles {
+		context := strings.ReplaceAll(d, "/Dockerfile", "")
+		contextParts := strings.Split(context, "/")
+		svcNameFromContext := contextParts[len(contextParts)-1]
+
+		// Check whether images contain service name derived from context
+		// as that's the best we can do in order to match a service to corresponding
+		// docker registry image. If no docker registry image was detected
+		// then we use service name as docker image name.
+		re := regexp.MustCompile(fmt.Sprintf(`.*\/%s`, svcNameFromContext))
+
+		for _, image := range analysis.Images {
+			if found := re.FindAllStringSubmatchIndex(image, -1); found != nil {
+				buildArtifacts[context] = image
+				break
+			} else {
+				buildArtifacts[context] = svcNameFromContext
+			}
+		}
+	}
+
+	artifacts := []*latest.Artifact{}
```

Comment:
Would be cool if this block (L:318 - L:327) is factored out into a func `e.g. setupImageArtefacts(...)`.